kuttl_tests.md

```diff
@@ -0,0 +1,69 @@
+Writing KUTTL test for an operator
+==================================
+
+Kuttl is a tool for testing Kubernetes operators. Each kuttl test suite
+consists of a series of test steps that might modify the state of the cluster
+or check its state. A kuttl test step will use all yaml files that begin with
+its numeric index. For example, if we had a file 00-deploy.yaml and
+a 00-assert.yaml, kuttl would apply the changes described by the file
+00-deploy.yaml and then wait until the state of the cluster matches the one
+found in 00-assert.yaml. If it's found, then the step is successful. If after
+the timeout, the state of the cluster does not match the assert, then the test
+fails. Kuttl also supports errors files, which will check that the cluster does
+not match the state described in the file.
+
+What should be tested for in a kuttl suite depends on the details of the
+operator itself. A good place to find more information is the controller.go
+file for the operator, there you should find
+hints on what resources does the operator create and/or manage. Another source
+of information would be to contact the people listed as approvers in the
+operator's `OWNERS` file. Below are some example of things to assert while
+writing the tests (they might not apply to all operators):
```

Comment:
i would also suggest mentioning the operator API as a source of info for which tests we can write eg https://github.com/openstack-k8s-operators/keystone-operator/blob/master/api/bases/keystone.openstack.org_keystoneapis.yaml to see what attributes the operator exposes and things like configOverwrite for the keystone example

Reply:
Good point, I've added it.